### DISCUSSION TITLE
ISP Port Blocking of 443

### DISCUSSION THREAD

**Discussion by IanK6449** [llamafilm/tesla-http-proxy-addon#81] [General]
So, it's taken me way too long to work this out but I was having a constant error around my public key not being available. I thought my port forwarding was correct and I could access my HA instance via port 443 externally. But, when I actually used a different network to my LAN I discovered that this wasn't the case. I looked into it a bit further and realised (durrr) that my ISP was blocking the inbound port 443. As such my public key was never accessible outside my LAN. A quick call to Aussie Broadband and everything is now working as it should be :-)

I thought this might prove a useful tip as I have seen many folks suffering the same public key issue.
The step by step guide on how to setup DuckDNS with both required domains was also pretty useful. But I couldn't get the DuckDNS add-on to download both certs until I uninstalled it, deleted the 2 key files in /ssh and then rebooted my machine and reinstalled the add-on. This may also help if anybody is having the same cert issues that I did.

Hope this helps anybody out there struggling like I did.

**Comment by baylanger** [comment]
This is very interesting, usually ISPs “bulk block” a port and don’t unblock for individual customers. You should review your ISP and give them 5* 🙃

For those facing the same issue but the ISP refuses to unblock your port, one solution is to use Cloudflare Argo Tunnel. This setup is already documented in the WiKi section.

**Reply by IanK6449** [reply]
Yeah Aussie Broadband are generally pretty good I've found. They can't unblock an individual port but did agree to unblock all the ports that were blocked. So. I probably have a bit of improvement to do security wise but at least I can now integrate HA with our Tesla :-)
